### PR TITLE
New action for navigation

### PR DESCRIPTION
**Pull request by Sakib75**
## PR Type

Feature

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] A changelog entry has been made.
- [ ] Version number has been updated.
- [ ] Required modules have been added to respective "requirements*.txt" files.
- [ ] Relevant Test Cases added to this description (below).
- [ ] (Team) Label with affected action categories and semver status.

## Overview
Added a new go to link action `go to link v2`. The will ignore any additional argument/experimental option/capacity when it first initiate Chrome. 
Example:
image
Options can be added manually
image

## Test Cases
- [TEST-8441](https://qa.automationsolutionz.com/Home/ManageTestCases/Edit/TEST-8441/#parentHorizontalTab2)

### CODE REVIEW

**Comment by sazid** [comment]
@Sakib75 for optional items, maybe we need to allow users to specify separate options for chrome, firefox and other browsers? Because some optional params and capabilities of one browser may be incompatible with the other browsers.

**Comment by Sakib75** [comment]
> @Sakib75 for optional items, maybe we need to allow users to specify separate options for chrome, firefox and other browsers? Because some optional params and capabilities of one browser may be incompatible with the other browsers.

@sazid Do you think the number of these params are limited? If yes, then we can filter them internally showing a log to the user.
This will keep the action dataset cleaner considering our intention to load the rows by default.

**Comment by Muntasib-creator** [comment]
In go-to-link action, if user adds any argument, node automatically removes other default arguments

File: Framework/MainDriverApi.py [hunk 1]
```diff
@@ -795,12 +795,11 @@ def zip_and_delete_tc_folder(
 
 def cleanup_driver_instances():  # cleans up driver(selenium, appium) instances
     try:  # if error happens. we don't care, main driver should not stop, pass in exception
-
-        if CommonUtil.teardown and shared.Test_Shared_Variables("selenium_driver"):
-            import Framework.Built_In_Automation.Web.Selenium.BuiltInFunctions as Selenium
-            driver = shared.Remove_From_Shared_Variables("selenium_driver")
-            if driver not in failed_tag_list:
-                Selenium.Tear_Down_Selenium()
+        import Framework.Built_In_Automation.Web.Selenium.BuiltInFunctions as Selenium
```
Comment (Muntasib-creator) [inline Framework/MainDriverApi.py:798]: why the condition is removed?

**Review by Muntasib-creator — CHANGES_REQUESTED** [review]

File: Framework/Built_In_Automation/Web/Selenium/BuiltInFunctions.py [hunk 2]
```diff
@@ -873,7 +872,6 @@ def firefoxheadless():
 
             selenium_driver.implicitly_wait(WebDriver_Wait)
             if not window_size_X and not window_size_Y:
-                selenium_driver.set_window_size(default_x, default_y)
```
Comment (Sakib75) [inline Framework/Built_In_Automation/Web/Selenium/BuiltInFunctions.py:876]: Need to confirm if `driver.maximize_window()` is working in headless mode

Comment (Sakib75) [reply Framework/Built_In_Automation/Web/Selenium/BuiltInFunctions.py:876]: @Muntasib-creator

File: Framework/MainDriverApi.py [hunk 3]
```diff
@@ -795,12 +795,11 @@ def zip_and_delete_tc_folder(
...
+        try:
+            Selenium.Tear_Down_Selenium()
```
Comment (Sakib75) [inline Framework/MainDriverApi.py:800]: | Scenario                                                       | Action | Test Result |
| -------------------------------------------------------------- | ------ | ----------- |
| Debug teardown - Normal                                        | Old    | Pass        |
| Debug teardown if previous session was forcefully closed       | Old    | Pass        |
| Deploy teardown if previous debug session was forefully closed | Old    | Pass        |
| Deploy teardown after running testcase in deploy mode          | Old    | Pass        |
| Debug continuity if Teardown is False                          | Old    | Pass        |
| In deploy initial Teardown is default                          | Old    | Pass        |
| Debug teardown - Normal                                        | New    | Pass        |
| Debug teardown if previous session was forcefully closed       | New    | Pass        |
| Deploy teardown if previous debug session was forefully closed | New    | Pass        |
| Deploy teardown after running testcase in deploy mode          | New    | Pass        |
| Debug continuity if Teardown is False                          | New    | Pass        |
| In deploy initial Teardown is default                          | New    | Pass        |

Comment (sazid) [reply Framework/MainDriverApi.py:800]: Can you modify the cells to say `v1` and `v2` instead of old and new? that will make it easier to understand.

**Review by sazid — APPROVED** [review]

**Pull request merged**